rclpy/test/listener.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import os
+import sys
+
+# this is needed to allow rclpy to be imported from the build folder
+sys.path.insert(0, os.getcwd())
+
+
+def listener_cb(msg, message_name, received_messages):
+    if message_name == 'String':
+        print('received: {})'.format(msg.data))
+    elif message_name == 'TransformStamped':
+        print('received: translation({},{},{})), rotation({},{},{},{})'.format(
+            msg.transform.translation.x,
+            msg.transform.translation.y,
+            msg.transform.translation.z,
+            msg.transform.rotation.x,
+            msg.transform.rotation.y,
+            msg.transform.rotation.z,
+            msg.transform.rotation.w))
+    elif message_name == 'Imu':
+        print('received: ({})'.format(msg.angular_velocity_covariance))
+    elif message_name == 'LaserScan':
+        print('received: ({})'.format(msg.intensities))
+    elif message_name == 'PointCloud2':
+        print('received: ({})'.format([f.name for f in msg.fields]))
```

Comment:
Each message should be able to print itself and make this conditionals obsolete:

    print('received: %r' % msg)

See ros2/rosidl#134 for the necessary `__repr__` method for Python message objects.